### PR TITLE
[search] Distinguish simple grouping from pre-defined search categories

### PR DESCRIPTION
**Pull request by pastk**
Follow-up to #7586

### CODE REVIEW

File: data/categories.txt [hunk 1]
```diff
@@ -1069,7 +1031,50 @@ sw:mafuta
...
+sk:Obchod
+sv:Butik|Affär
+sw:Duka
+th:ร้านค้า
+tr:Mağaza
+uk:4Магазин|Крамниця
+vi:Cửa hàng
+zh-Hans:商店
+zh-Hant:1購物|店鋪|商店|雜貨店|便利商店
+ca:Botiga|Tenda
```
Comment (biodranik) [inline data/categories.txt:1075]: Sort if it was moved?

File: data/categories.txt [hunk 2]
```diff
@@ -4721,43 +4721,6 @@ zh-Hans:赌博
 zh-Hant:賭博
 
 amenity-gambling|@gambling
-en:Gambling
```
Comment (biodranik) [inline data/categories.txt:4719]: Did you test if it works like this?

**Review by biodranik — APPROVED** [review]
LGTM, @vng PTAL

File: data/categories.txt (hunk 2 above)
Comment (pastk) [reply data/categories.txt:4719]: yeap, it works
and there are other "empty" entries like this, e.g. `shop|@category_shop`

File: data/categories.txt (hunk 1 above)
Comment (pastk) [reply data/categories.txt]: sorted!

**Review by vng — APPROVED** [review]

**Pull request merged**